x-pack/legacy/plugins/rollup/public/plugin.tsx

```diff
@@ -0,0 +1,155 @@
+import { setNotifications, setFatalErrors } from './kibana_services';
+
+export interface RollupPluginSetupDependencies {
+  __LEGACY: {
+    aggTypeFilters: AggTypeFilters;
+    aggTypeFieldFilters: AggTypeFieldFilters;
+    editorConfigProviders: EditorConfigProviderRegistry;
+    addSearchStrategy: (searchStrategy: SearchStrategyProvider) => void;
+    management: ManagementSetup;
+    addBadgeExtension: (badgeExtension: any) => void;
+    addToggleExtension: (toggleExtension: any) => void;
+  };
+  home?: HomePublicPluginSetup;
+}
+
+export interface RollupPluginStartDependencies {
+  management: ManagementStart;
+}
+
+export class RollupPlugin implements Plugin {
+  setup(
+    core: CoreSetup,
+    {
+      __LEGACY: {
+        aggTypeFilters,
+        aggTypeFieldFilters,
+        editorConfigProviders,
+        addSearchStrategy,
+        management: managementLegacy,
+        addBadgeExtension,
+        addToggleExtension,
+      },
+      home,
+    }: RollupPluginSetupDependencies
+  ) {
+    setFatalErrors(core.fatalErrors);
+    addBadgeExtension(rollupBadgeExtension);
+    addToggleExtension(rollupToggleExtension);
+
+    const isRollupIndexPatternsEnabled = core.uiSettings.get(CONFIG_ROLLUPS);
+
+    if (isRollupIndexPatternsEnabled) {
+      managementLegacy.indexPattern.creation.add(RollupIndexPatternCreationConfig);
+      managementLegacy.indexPattern.list.add(RollupIndexPatternListConfig);
+      addSearchStrategy(getRollupSearchStrategy(core.http.fetch));
+      initAggTypeFilter(aggTypeFilters);
+      initAggTypeFieldFilter(aggTypeFieldFilters);
+      initEditorConfig(editorConfigProviders);
+    }
+
+    if (home) {
+      home.featureCatalogue.register({
+        id: 'rollup_jobs',
+        title: 'Rollups',
+        description: i18n.translate('xpack.rollupJobs.featureCatalogueDescription', {
+          defaultMessage:
+            'Summarize and store historical data in a smaller index for future analysis.',
+        }),
+        icon: 'indexRollupApp',
+        path: `#${CRUD_APP_BASE_PATH}/job_list`,
+        showOnHomePage: true,
+        category: FeatureCatalogueCategory.ADMIN,
+      });
+    }
+  }
+
+  start(core: CoreStart, { management }: RollupPluginStartDependencies) {
+    setHttp(core.http);
+    setNotifications(core.notifications);
+    setEsBaseAndXPackBase(core.docLinks.ELASTIC_WEBSITE_URL, core.docLinks.DOC_LINK_VERSION);
+
+    const esSection = management.sections.getSection('elasticsearch');
+
+    const I18nContext = core.i18n.Context;
+
+    esSection!.registerApp({
+      id: 'rollup_jobs',
+      title: i18n.translate('xpack.rollupJobs.appTitle', { defaultMessage: 'Rollup Jobs' }),
+      order: 3,
+      mount(params) {
+        params.setBreadcrumbs([
+          {
+            text: i18n.translate('xpack.rollupJobs.breadcrumbsTitle', {
+              defaultMessage: 'Rollup Jobs',
+            }),
+          },
+        ]);
+
+        render(
```

Comment:
Minor; see the NP [application conventions](https://github.com/elastic/kibana/blob/master/src/core/CONVENTIONS.md#applications). It is a minor difference but the convention recommends importing a single `renderApp` function that does the `react-dom` `render` and `unmountComponentAtNode`.

Reply:
Nice catch! I moved the template to a separate file.